front/app/components/admin/Graphs/BarChart/index.tsx

```diff
@@ -0,0 +1,165 @@
+  isAnimationActive: false;
+  cursor: { fill: string };
+}
+
+interface Props {
+  width?: string | number;
+  height?: string | number;
+  data?: Data | null | Error;
+  mapping?: Mapping;
+  layout?: Layout;
+  margin?: Margin;
+  bars?: BarProps;
+  xaxis?: AxisProps;
+  yaxis?: AxisProps;
+  renderLabels?: (props: RenderLabelsProps) => React.ReactNode;
+  renderTooltip?: (props: RenderTooltipProps) => React.ReactNode;
+  emptyContainerContent?: React.ReactNode;
+  className?: string;
+  innerRef?: any;
+}
+
+const BarChart = ({
+  width,
+  height,
+  data,
+  mapping,
+  layout = 'vertical',
+  margin,
+  bars,
+  xaxis,
+  yaxis,
+  renderLabels,
+  renderTooltip,
+  emptyContainerContent,
+  className,
+  innerRef,
+}: Props) => {
+  const {
+    chartLabelSize,
+    chartLabelColor,
+    animationBegin,
+    animationDuration,
+    newBarFill,
+    barHoverColor,
+  }: any = useTheme();
+
+  const noData = isNilOrError(data) || data.every(isEmpty) || data.length <= 0;
+
+  if (noData) {
+    return (
+      <NoDataContainer>
+        {emptyContainerContent ? (
+          <>{emptyContainerContent}</>
+        ) : (
+          <FormattedMessage {...messages.noData} />
+        )}
+      </NoDataContainer>
+    );
+  }
+
+  const { length, fill } = parseMapping(mapping);
+  const rechartsLayout = getRechartsLayout(layout);
+  const parsedBarProps = parseBarProps(newBarFill, bars);
+
+  return (
+    <ResponsiveContainer className={className} width={width} height={height}>
+      <_BarChart
+        data={data}
+        layout={rechartsLayout}
+        margin={margin}
+        ref={innerRef}
+      >
+        {renderTooltip &&
+          renderTooltip({
+            isAnimationActive: false,
+            cursor: { fill: barHoverColor },
+          })}
+
+        <Bar
+          dataKey={length}
+          animationDuration={animationDuration}
+          animationBegin={animationBegin}
+          {...parsedBarProps}
+        >
+          {renderLabels &&
+            renderLabels({ fill: chartLabelColor, fontSize: chartLabelSize })}
+
+          {fill &&
+            data.map((row, index) => (
+              <Cell
+                key={`cell-${index}`}
+                fill={row[fill] || parsedBarProps.fill}
+              />
+            ))}
+        </Bar>
+
+        <XAxis
+          dataKey={layout === 'vertical' ? 'name' : length}
+          type={layout === 'vertical' ? 'category' : 'number'}
+          stroke={chartLabelColor}
+          fontSize={chartLabelSize}
+          tick={{ transform: 'translate(0, 7)' }}
+          {...xaxis}
+        />
+        <YAxis
+          dataKey={layout === 'horizontal' ? 'name' : length}
+          type={layout === 'horizontal' ? 'category' : 'number'}
+          stroke={chartLabelColor}
+          fontSize={chartLabelSize}
+          {...yaxis}
+        />
+      </_BarChart>
+    </ResponsiveContainer>
+  );
+};
+
+export default BarChart;
+
+export const DEFAULT_MARGIN: Margin = {
```

Comment:
Should this be more concrete, like `DEFAULT_BAR_CHART_MARGIN` maybe?

Reply:
I thought that since it lives in the `BarChart` file you can assume it relates to the bar chart only, but if you think it's not clear I can change it